src/composables/useRoutes.ts

```diff
@@ -62,30 +62,90 @@ export const useRoutes = () => {
   };
 
   /**
-   * Generate an array of RouteListDay objects grouped by date from the given routes.
-   *
-   * @param routes - The array of RouteItem objects to process.
-   * @return The array of RouteListDay objects grouped by date.
+   * Creates an array of RouteListDay objects for each day between specified
+   * dates. Including end date and excluding start date.
+   * Fills in data from routes array based on date and direction.
+   * If data is empty for given day/route, it will create an empty route.
+   * @param {Date} startDate - The start date of the date range.
+   * @param {Date} endDate - The end date of the date range.
+   * @param {RouteItem[]} routes - The array logged routes.
+   * @return {RouteListDay[]} The array representing days with routes.
    */
-  const getDays = (routes: RouteItem[] | null | undefined): RouteListDay[] => {
-    const dayArray: RouteListDay[] = [];
-    if (!routes) return dayArray;
-
-    routes?.forEach((route) => {
-      const routeDate = route.date;
-      let day: RouteListDay | undefined = dayArray.find((day) => {
-        return day.date === routeDate;
-      });
-
-      if (!day) {
-        day = { date: routeDate, routes: [] };
-        dayArray.push(day);
+  const createDaysArrayWithRoutes = (
+    startDate: Date,
+    endDate: Date,
+    routes: RouteItem[],
+  ): RouteListDay[] => {
+    const numberOfDays = date.getDateDiff(endDate, startDate, 'days');
+    const days = [] as RouteListDay[];
+    if (routes) {
+      for (let i = 0; i < numberOfDays; i++) {
+        const currentDate = date.subtractFromDate(endDate, { days: i });
+
+        // For any given day, get data from routes or create an empty route.
+        const fromWork = getRouteByDateAndDirection(
+          routes,
+          currentDate,
+          TransportDirection.fromWork,
+        );
+        const toWork = getRouteByDateAndDirection(
+          routes,
+          currentDate,
+          TransportDirection.toWork,
+        );
+        days.push({
+          id: date.formatDate(currentDate, 'YYYY-MM-DD'),
```

Comment:
Date format literal string repeat  across code, you can make it as const.

Reply:
Updated. Used a local const `routeDateFormat`.